kafka-rest/src/main/java/io/confluent/kafkarest/controllers/ReplicaManagerImpl.java

```diff
@@ -84,15 +88,17 @@ public CompletableFuture<List<PartitionReplica>> searchReplicasByBrokerId(
               return KafkaFutures.toCompletableFuture(result.values().get(brokerId));
             })
         .thenCompose(
-            logDirs ->
-                CompletableFutures.allAsList(
-                    logDirs.values().stream()
-                        .flatMap(logDir -> logDir.replicaInfos.keySet().stream())
-                        .map(
-                            partition ->
-                                getReplica(
-                                    clusterId, partition.topic(), partition.partition(), brokerId))
-                        .collect(Collectors.toList())))
+            logDirs -> {
+              log.debug("Describe log dirs {} ", logDirs);
```

Comment:
Did you double check this actually prints out the full list of logDirs?

Reply:
Yes, double checked, this will